### ISSUE TITLE
`solveset_real(x**3+1, x)` returns `EmptySet()`

### ISSUE THREAD

**Issue by gxyd** [sympy/sympy#9778]
```
>>> x = Symbol('x', real=True)
>>> solveset_real(x**3+1,x)
EmptySet()          # should return FiniteSet(-1)

>>> solveset_complex(x**3+1,x)   # though this works
{-1, 1/2 - sqrt(3)*I/2, 1/2 + sqrt(3)*I/2}
```

As mentioned by @shivamvats [here](https://github.com/sympy/sympy/pull/9762#issuecomment-127285393)

**Comment by aktech** [comment]
Any one fixing this should note:
This behaviour is due to:

```
In [8]: FiniteSet((-1)**(S(1)/3)).intersect(S.Reals)
Out[8]: EmptySet()
```
As mentioned [here](https://github.com/sympy/sympy/pull/9762#issuecomment-127399912).

**Comment by gxyd** [comment]
> This behaviour is due to:
In [8]: FiniteSet((-1)**(S(1)/3)).intersect(S.Reals)
Out[8]: EmptySet()

I think this behavior is correct. Since in all deal with the `principal value` of `(-1)**(S(1)/3)` which is a `non-real` number in itself. Also  `(-1)**(S(1)/2)` gives the principal value `only` i.e. `i (imaginary unit)`.

**Comment by aktech** [comment]
> I think this behavior is correct.

Yes, Indeed. I don't mean that's a wrong behaviour, that's just the point in the code, where we missed our solution.

**Comment by hargup** [comment]
> This behaviour is due to:
In [8]: FiniteSet((-1)**(S(1)/3)).intersect(S.Reals)
Out[8]: EmptySet()

This wrong as by definition [#1] of cube roots on real numbers which defines it as a one-to-one mapping from reals to reals.  Our current behaviour leads to a situation where we cannot even determine if one of the roots of `x**3 +y` is real given `y` is real.
```
In [66]: y = Symbol('y', real=True)

In [67]: list(map(lambda r: r.is_real, roots(x**3 + y, x).keys())
   ....: )
Out[67]: [None, None, None]
```

Also if we want to return the principal value of the cube root then we should return `1/2 + sqrt(3)*I/2`, keeping `(-1)**(S(1)/3)` unevaluted doesn't make sense.

[#1]: http://mathworld.wolfram.com/CubeRoot.html

**Comment by gxyd** [comment]
Fixed by #9804 .
